src/main/components/form/validator.ts

```diff
@@ -189,10 +189,12 @@ export const isValidInteger: Validator = value => {
     return 'invalid';
   }
 
-  if (/^\d+$/.test(value as string)) {
-    return;
-  } else {
-    return 'invalid';
+  if (/^\D+$/.test(value as string) || /^\d+\D+$/.test(value as string)) {
+    return 'notANumber';
+  }
+
+  if (/^\D+$/.test(value as string) || /^\d+[^0-9.]+$/.test(value as string)) {
```

Comment:
remove first condition and move the second condition to above if statement

Reply:
Fixed duplicate